src/main/java/uk/gov/hmcts/darts/event/service/impl/CleanupCurrentFlagEventProcessorImpl.java

```diff
@@ -19,36 +20,26 @@
 @Getter
 @RequiredArgsConstructor
 @Slf4j
+@Component
 public class CleanupCurrentFlagEventProcessorImpl implements CleanupCurrentFlagEventProcessor {
-    private final Integer batchSize;
+    private final Integer batchSize = 20;
     private final EventRepository eventRepository;
     private final HearingRepository hearingRepository;
     private final UserIdentity userIdentity;
 
     @Override
-    public List<Integer> processCurrentEvent() {
-        List<Integer> processedEventIdLst = new ArrayList<>();
-        log.debug("Batch size to process event ids for {}", batchSize);
-        List<Integer> eventEntityReturned = eventRepository.findCurrentEventIdsWithDuplicates(batchSize);
-        if (log.isDebugEnabled()) {
-            log.debug("Event ids being processed {}", eventEntityReturned
-                .stream()
-                .map(Object::toString)
-                .collect(Collectors.joining(",")));
+    @Async
```

Comment:
Would this reuse the executor defined in `AsyncEventsConfig`? You might find this will be competing for available pool threads with other parts of the code, including some potentially long-running processes / batch jobs.

Consider configuring a thread pool dedicated to processing these events, configured appropriately for the expected event frequency.

Reply:
Agreed I have created a new taskExecutor just for this